lib/reservoir-sampler.js

```diff
@@ -94,12 +105,16 @@ ReservoirSampler.prototype._read = function() {
         _id: 1
       },
       sort: this.sort,
-      limit: 10000
+      limit: RESERVOIR_SAMPLE_LIMIT
     })
       .stream()
-      .pipe(createReservoir(this.size))
-      .pipe(_idToDocument(this.db, this.collectionName, {
-        fields: this.fields
+      .pipe(es.through(function write(obj) {
```

Comment:
Up to you, but this might be a bit more clear if it was instead:
```
.pipe(es.map(function (doc, cb) {
  cb(null, doc._id);
})
```

It took me a second to realize that you were doing a map here on first read :)